### PR TITLE
[2.0.x] Apply fixes for DUE

### PR DESCRIPTION
**Pull request by thinkyhead**
Alternative to #7882.

If F_CPU is greater than 1000 it can be evenly divided by 8. Over 10000, 16; over 100000, 32; over 1 million, 64; etc. so there's no need to enforce `double` on `HAL_TIMER_RATE`. Rather, code which uses these values should decide whether casts are needed.

- 1,000 / 8 = 125
- 10,000 / 16 = 625
- 100,000 / 32 = 3,125
- 1,000,000 / 64 = 15,625
- 10,000,000 / 128 = 78,125
- etc.

### CODE REVIEW

**Comment by p3p** [comment]
I can't test Due firmware but it looks like it should fix the issue (it compiles now at least) though in testing that I had to add `lib_ignore  = TMC26XStepper` to the platformio.ini DUE section .. looks like they don't like each other. I think we need to get at least 1 default config test running for each platform in Travis so we at least know when things don't work at all ^^

File: Marlin/src/module/stepper.h [hunk 1]
```diff
@@ -303,10 +303,9 @@ class Stepper {
 
       #ifdef CPU_32_BIT
         // In case of high-performance processor, it is able to calculate in real-time
-        timer = (uint32_t)(HAL_STEPPER_TIMER_RATE) / step_rate;
-        if (timer < (HAL_STEPPER_TIMER_RATE / (STEP_DOUBLER_FREQUENCY * 2))) { // (STEP_DOUBLER_FREQUENCY * 2 kHz - this should never happen)
-          timer = (HAL_STEPPER_TIMER_RATE / (STEP_DOUBLER_FREQUENCY * 2));
-        }
+        constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
```
Comment (tcm0116) [inline Marlin/src/module/stepper.h:306]: Defining `MIN_TIME_PER_STEP` as a `constexpr` breaks the LPC1768 build with the following error:

```
Marlin/src/module/stepper.h: In static member function 'static uint32_t Stepper::calc_timer(uint32_t)':
Marlin/src/module/stepper.h:306:104: error: the value of 'SystemCoreClock' is not usable in a constant expression
constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
^
```

Changing `constexpr` to just `const` works.

**Comment by apballard** [comment]
I'm also getting errors based on this change:

In file included from Marlin/src/Marlin.cpp:36:0:
Marlin/src/module/stepper.h: In static member function 'static uint32_t Stepper::calc_timer(uint32_t)':
Marlin/src/module/stepper.h:306:104: error: the value of 'SystemCoreClock' is not usable in a constant expression

constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
^
In file included from frameworks/CMSIS/LPC1768/include/LPC17xx.h:104:0,
from Marlin/src/inc/../HAL/HAL_LPC1768/fastio.h:37,
from Marlin/src/inc/../HAL/HAL_LPC1768/HAL.h:59,
from Marlin/src/inc/../HAL/HAL.h:47,
from Marlin/src/inc/MarlinConfig.h:33,
from Marlin/src/Marlin.h:29,
from Marlin/src/Marlin.cpp:31:
frameworks/CMSIS/LPC1768/include/system_LPC17xx.h:31:17: note: 'uint32_t SystemCoreClock' is not const
extern uint32_t SystemCoreClock; /*!< System Clock Frequency (Core Clock) */
^~~~~~~~~~~~~~~
*** [.pioenvs/Re-ARM/src/src/Marlin.o] Error 1
In file included from Marlin/src/core/../module/temperature.h:39:0,
from Marlin/src/core/utility.cpp:26:
Marlin/src/core/../module/stepper.h: In static member function 'static uint32_t Stepper::calc_timer(uint32_t)':
Marlin/src/core/../module/stepper.h:306:104: error: the value of 'SystemCoreClock' is not usable in a constant expression
constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
^
In file included from frameworks/CMSIS/LPC1768/include/LPC17xx.h:104:0,In file included from Marlin/src/Marlin.cpp:36:0:
Marlin/src/module/stepper.h: In static member function 'static uint32_t Stepper::calc_timer(uint32_t)':
Marlin/src/module/stepper.h:306:104: error: the value of 'SystemCoreClock' is not usable in a constant expression

constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
^
In file included from frameworks/CMSIS/LPC1768/include/LPC17xx.h:104:0,
from Marlin/src/inc/../HAL/HAL_LPC1768/fastio.h:37,
from Marlin/src/inc/../HAL/HAL_LPC1768/HAL.h:59,
from Marlin/src/inc/../HAL/HAL.h:47,
from Marlin/src/inc/MarlinConfig.h:33,
from Marlin/src/Marlin.h:29,
from Marlin/src/Marlin.cpp:31:
frameworks/CMSIS/LPC1768/include/system_LPC17xx.h:31:17: note: 'uint32_t SystemCoreClock' is not const
extern uint32_t SystemCoreClock; /*!< System Clock Frequency (Core Clock) */
^~~~~~~~~~~~~~~
*** [.pioenvs/Re-ARM/src/src/Marlin.o] Error 1
In file included from Marlin/src/core/../module/temperature.h:39:0,
from Marlin/src/core/utility.cpp:26:
Marlin/src/core/../module/stepper.h: In static member function 'static uint32_t Stepper::calc_timer(uint32_t)':
Marlin/src/core/../module/stepper.h:306:104: error: the value of 'SystemCoreClock' is not usable in a constant expression
constexpr uint32_t MIN_TIME_PER_STEP = (HAL_STEPPER_TIMER_RATE) / ((STEP_DOUBLER_FREQUENCY) * 2);
^
In file included from frameworks/CMSIS/LPC1768/include/LPC17xx.h:104:0,

File: Marlin/src/module/stepper.h (hunk 1 above)
Comment (TGMods) [reply Marlin/src/module/stepper.h:306]: @tcm0116 @thinkyhead 
I can confirm that this breaks the LPC1768 build for my Re-ARM and that changing constexpr to just const works.

**Pull request merged**